content/blog/_posts/2021-04-rusty-workflows.md

```diff
@@ -0,0 +1,725 @@
+---
+title: GitHub Actions best practices for Rust projects
+author:
+    name: "Nick Mosher"
+    github: "nicholastmosher"
+description: "How I cleaned up Rust CI boilerplate using the GitHub Actions build matrix"
+date: 2021-04-15
+slug: github-actions-best-practices
+url: /blog/2021/04/github-actions-best-practices
+twitter-card: summary_large_image
+code:
+    height: 9000
+---
+
+[Fluvio] is a high-performance distributed streaming platform written in Rust.
+As a fairly large project, we have a lot of build configurations and testing
+scenarios that we automate in order to make sure we don't break things by
+accident. We've been using GitHub Actions in order to run our CI/CD workflows,
+but as we've grown, things have naturally gotten messy over time. This week,
+I took some time to re-visit our workflow definitions to clean things up and
+try to increase our team's productivity. I specifically want to talk about
+two main improvements I worked on:
+
+[Fluvio]: https://github.com/infinyon/fluvio/
+
+- Consolidating multiple jobs using the build matrix
+- Setting up `sccache` to improve our building and testing speed
+  - We actually had already set up `sccache` but it was misconfigured.
+    I'll talk about how to check that everything is set up properly.
+
+The first half of this post should be generally useful for anybody who needs
+to use GitHub Actions and wants to learn more about the build matrix. The
+second half should be useful to Rust developers who want a good starting point
+for a solid CI setup.
+
+## Using the GitHub workflows build matrix
```

Comment:
A title such as `From Makefiles to Github Action Workflows` would catch my attention. (not sure if this is the point of the section... as I haven't read it yet).